types/sharedb/index.d.ts

```diff
@@ -93,6 +103,20 @@ declare class sharedb extends EventEmitter {
     on(event: 'timing', callback: (type: string, time: number, request: any) => void): this;
     on(event: 'submitRequestEnd', callback: (error: Error, request: SubmitRequest) => void): this;
     on(event: 'error', callback: (err: Error) => void): this;
+    on(event: 'send', callback: (agent: Agent, context: SubmitContext
+        | ApplyContext
+        | CommitContext
+        | ConnectContext
+        | DocContext
+        | OpContext
+        | QueryContext
+        | ReadSnapshotsContext
+        | ReceiveContext
+        | ReplyContext
+        /**
+         * For errors, we'll send back the request body (ReceiveContext.data).
+         */
+        | {[key: string]: any}) => void): this;
```

Comment:
For errors, the first arg is still the Agent, as shown in the first source code link above.

Reply:
I think what you're saying is we need to add multiple send signatures, not just for the happy case.